source/libraries/ContractInterfacesGenerator.ts

```diff
@@ -74,7 +74,7 @@ export class ContractInterfaceGenerator {
             const data = encodeMethod(abi, parameters);
             let gas = await this.connector.ethjsQuery.estimateGas(Object.assign({ to: this.address, from: from, data: data }, attachedEth ? { value: attachedEth } : {} ));
             // This is to address an observed bug while running against geth where occasionally gas estimates are lower than required: https://github.com/ethereum/go-ethereum/issues/15896
-            gas = gas.add(gas.div(new BN(10)));
+            gas = BN.min(new BN(6950000), gas.add(gas.div(new BN(10))));
```

Comment:
Minor: might be nice if this was a constant, perhaps it could come out of the the configuration object, since it's already involved with the network and gasPrice.

Reply:
That's a good idea, I'll do it there so its less magic and more expicit